pkgs/development/interpreters/perl/interpreter.nix

```diff
@@ -67,6 +67,9 @@ stdenv.mkDerivation (rec {
     ++ lib.optional ((lib.versions.majorMinor version) == "5.38") ./no-sys-dirs-5.38.0.patch
     ++ lib.optional ((lib.versions.majorMinor version) == "5.40") ./no-sys-dirs-5.40.0.patch
 
+    # Fix compilation on platforms with only a C locale: https://github.com/Perl/perl5/pull/22569
+    ++ lib.optional (version == "5.40.0") ./fix-build-with-only-C-locale-5.40.0.patch
```

Comment:
Do we need to vendor the patch, or can we fetchpatch https://github.com/Perl/perl5/commit/bd0ab509f890a6638bd5033ef58526f8c74f7e4b.patch?

Reply:
Likely due to:

```
# Note: this package is used for bootstrapping fetchurl, and thus
# cannot use fetchpatch! All mutable patches (generated by GitHub or
# cgit) that are needed here should be included directly in Nixpkgs as
# files.
```

Reply:
I sometimes wonder why Nix's [`builtin:fetchurl`](https://github.com/NixOS/nix/blob/master/src/libexpr/fetchurl.nix#L19) isn't used in situations like this.